Automattic/wp-calypso
# Send some users to the concierge upsell page after plan purchase

#### Changes proposed in this Pull Request

This adds the logic to conditionally show the new concierge session upsell page to a user after they purchase the Blogger, Personal, or Premium plan.
It's currently hidden behind a feature flag which is only enabled on development.
This upsell will be tested against the GSuite upsell.
The initial percentage of users to allocate to the test group might change.

#### Testing instructions

Start calypso locally with this branch.

1. Put yourself into the `show` test group.
<img width="599" alt="screen shot 2018-12-17 at 4 01 59 pm" src="https://user-images.githubusercontent.com/690843/50123444-8591bb80-0215-11e9-919f-7fac915569e6.png">

2. Create a new site (**don't** choose "_Sell products or collect payments_" for the "_What’s the primary goal..._" question).
3. Add a Premium plan and domain to the cart.
<img width="327" alt="screen shot 2018-12-17 at 5 11 10 pm" src="https://user-images.githubusercontent.com/690843/50125847-61d37300-021f-11e9-84c7-86ce11e41093.png">

4. Purchase the items.
5. After purchase, confirm you're shown the Concierge upsell page.
<img width="788" alt="screen shot 2018-12-17 at 5 12 06 pm" src="https://user-images.githubusercontent.com/690843/50125784-33559800-021f-11e9-9f66-ec07c7c6917b.png">

6. Click the button: "Reserve a call for $99.00". You should be taken to the checkout.
<img width="336" alt="screen shot 2018-12-17 at 5 13 25 pm" src="https://user-images.githubusercontent.com/690843/50125808-410b1d80-021f-11e9-955a-a5845e905034.png">

7. Purchase the session
8. Confirm you're taken to the checklist.
<img width="828" alt="screen shot 2018-12-17 at 5 13 46 pm" src="https://user-images.githubusercontent.com/690843/50125814-4700fe80-021f-11e9-905f-8ec53b32e6ed.png">

-------------

Next, put yourself into the `skip` test group and run through steps 2-4 again.
This time you should be shown the G Suite upsell instead. No need to purchase GSuite, just confirm the upsell page is shown.
<img width="782" alt="screen shot 2018-12-17 at 5 22 36 pm" src="https://user-images.githubusercontent.com/690843/50126112-6a787900-0220-11e9-9fcd-037f41f55997.png">
